R/site.R

```diff
@@ -68,3 +68,48 @@ rmd_needs_render <- function(path) {
 rmd_hash <- function(path) {
   digest::digest(path, file = TRUE, algo = "xxhash64")
 }
+
+hugodown_site <- function(input = ".", output_format = NULL) {
+  # read config and output directory
+  config <- hugo_config(input)
+  output_dir <- hugo_config_str(config, "publishdir")
+
+  list(
+    # name for site (used as default name for publishing)
+    name = hugo_config_str(config, "title"),
+
+    # output directory (e.g. "public")
+    output_dir = output_dir,
+
+    # render_site callback
+    render = function(input_file, output_format, envir, quiet, ...) {
+
+      # Render the entire site
+      if (is.null(input_file)) {
+        hugo_build(input)
+        if (!quiet) {
+          cat("Hugo site built:", file.path(input, output_dir))
+        }
+
+        # render just a single file (this hook is for doing incremental
+        # renders of the site based on a "Knit" in the IDE). Since
+        # hugodown already handles this well w/ just delegate here to
+        # rmarkdown::render.
+      } else {
+        rmarkdown::render(
```

Comment:
This is run in a clean session, because the IDE launches a session just for this task, right?

Reply:
Yes it will be in a clean session when run from the IDE (it won't be if `render_site()` is called from the R console)